.github/workflows/docs.yml

```diff
@@ -0,0 +1,47 @@
+name: API Reference
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
+
+jobs:
+  build:
+    if: github.actor != 'dependabot[bot]' || github.event_name != 'pull_request'
+    runs-on: ubuntu-latest
+    permissions:
+      deployments: write
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Set up Java 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+      - name: Set up Gradle
+        uses: gradle/gradle-build-action@v2
+
+      - name: Build Documentation
+        run: ./gradlew dokkaHtml
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: eu-west-2
+          role-to-assume: arn:aws:iam::${{ secrets.ABLY_AWS_ACCOUNT_ID_SDK }}:role/ably-sdk-builds-ably-chat-js
```

Comment:
Should we have a new role for Kotlin?

Reply:
Sure, will do, waiting for permissions to create new IAM role  :)